public/app/features/datasources/state/actions.ts

```diff
@@ -248,9 +270,23 @@ export function loadDataSourcePlugins(): ThunkResult<void> {
 }
 
 export function updateDataSource(dataSource: DataSourceSettings) {
-  return async (dispatch: (dataSourceSettings: ThunkResult<Promise<DataSourceSettings>>) => DataSourceSettings) => {
-    await api.updateDataSource(dataSource);
+  return async (
+    dispatch: (
+      dataSourceSettings: ThunkResult<Promise<DataSourceSettings>> | { payload: any; type: any }
```

Comment:
I guess `type` can only be a `string`? Probably we could also use `unknown` instead of `any` for `payload` as well.